addrmgr/addrmanager.go

```diff
@@ -844,6 +844,20 @@ func (a *AddrManager) find(addr *wire.NetAddress) *KnownAddress {
 	return a.addrIndex[NetAddressKey(addr)]
 }
 
+// GetLastAttempt retrieves an address' last attempt time.
+func (a *AddrManager) GetLastAttempt(addr *wire.NetAddress) time.Time {
+	a.mtx.Lock()
+	defer a.mtx.Unlock()
+
+	ka := a.find(addr)
```

Comment:
What about instead accepting the `KnownAddress` directly? Alternative, can we instead protect `LastAttempt` with a mutex?

Reply:
Both would work. Passing in `KnownAddress` would be easier since the addressmanager already has a mutex, but protecting `LastAttempt` with a mutex would be better stylistically. I think I'll do the latter

Reply:
I think I'll need to dig into the code a bit more. It seems that a lot of the members of `KnownAddress` can be racy, and the same goes for `NetAddress`.  Think I need to figure out the best way to patch in one sweep.

Reply:
Has there been any progress on this?

Reply:
Still need to figure out all the fields that race